### PR TITLE
Issues #38 and #39 fixes - Work in progress

### PR DESCRIPTION
**Pull request by latin-programmer**
Hi pongstylin, 

I've been working on this off and on and I got stuck after I stopped getting errors. Classic up seems to be back in working order. I am no longer getting console errors for face-off and chaos apps, but those two are still wonky. 

Overall, my goal was to per our discussion to bring the "data" property concept to the local transport and have the apps match it. This took care of the errors but I am still having some issues getting the bots to talk to the human player properly (and the worker sometimes just stops receiving messages). 

Of course, I want everything to be back in 100% working order but I decided it would be best to show you what I have and get some guidance on what I'm missing. Heck, you might even see one or two lines of code that would get them back in order. -El Cid

### CODE REVIEW

File: src/chaos-app.js [hunk 1]
```diff
@@ -10,8 +10,9 @@ Tactics.App = (function ($, window, document) {
     {assignment:[5, 1], type:'Knight'},
     {assignment:[6, 1], type:'Knight'},
   ];
-  var gameData = {
```
Comment (pongstylin) [inline src/chaos-app.js]: @latin-programmer Nesting the data inside of a "data" tag shouldn't be necessary and should be reverted.  The so-called "gameData" variable would probably be better named as "gameStateData" since it will ultimately be consumed by the "tactics/GameState.js" class when loading an instance.

File: src/chaos-app.js [hunk 2]
```diff
@@ -10,8 +10,9 @@ Tactics.App = (function ($, window, document) {
...
-    type: 'Chaos',
+  var gameData = { data: {} };
```
Comment (pongstylin) [inline src/chaos-app.js]: @latin-programmer Nesting the data inside of a "data" tag shouldn't be necessary and should be reverted. The so-called "gameData" variable would probably be better named as "gameStateData" since it will ultimately be consumed by the "tactics/GameState.js" class when loading an instance.

File: src/classic-app.js [hunk 3]
```diff
@@ -20,7 +20,8 @@ Tactics.App = (function ($, window, document) {
     {assignment:[6, 2], type:'Knight'},
     {assignment:[9, 2], type:'Scout'},
   ];
-  var gameData = {
+  var gameData = { data: {} };
```
Comment (pongstylin) [inline src/classic-app.js]: @latin-programmer Same here.  Nesting is bad.

File: src/faceoff-app.js [hunk 4]
```diff
@@ -10,7 +10,8 @@ Tactics.App = (function ($, window, document) {
     {assignment:[5, 2], type:'Knight'},
     {assignment:[6, 2], type:'Knight'},
   ];
-  var gameData = {
+  var gameData = { data: {} };
```
Comment (pongstylin) [inline src/faceoff-app.js]: @latin-programmer Same here, nesting is bad.

File: src/tactics/Bot.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ export default class {
      * Detect when it is time to start this bot's turn.
      */
     let startTurnListener = event => {
-      if (event.teamId === this.team.id)
+      if (event.data.teamId === this.team.id)
```
Comment (pongstylin) [inline src/tactics/Bot.js]: Good!

File: src/tactics/LocalTransport.js [hunk 6]
```diff
@@ -27,14 +27,14 @@ export default class LocalTransport {
    */
   static createGame(gameData) {
     let transport = new LocalTransport();
-    transport._post({ type:'create', data:gameData });
+    transport._post({ type:'create', data:gameData.data });
```
Comment (pongstylin) [inline src/tactics/LocalTransport.js]: @latin-programmer Once you stop nesting the data, you can revert this change as well.

File: src/tactics/LocalTransport.js [hunk 7]
```diff
@@ -27,14 +27,14 @@ export default class LocalTransport {
...
 
     return transport;
   }
 
   static loadGame(gameData) {
     let transport = new LocalTransport();
-    transport._post({ type:'load', data:gameData });
+    transport._post({ type:'load', data:gameData.data });
```
Comment (pongstylin) [inline src/tactics/LocalTransport.js]: @latin-programmer This can be reverted as well.

File: src/tactics/LocalTransport.js [hunk 8]
```diff
@@ -148,28 +148,29 @@ export default class LocalTransport {
...
-        .on('startGame', ({data}) => {
-          this._data = data;
+        .on('startGame', ({data:stateData}) => {
+          this._data.state = stateData;
         })
-        .on('startTurn', ({turnId, teamId}) => {
-          Object.assign(this._data, {
-            currentTurnId: turnId,
-            currentTeamId: teamId,
+        .on('startTurn', ({ data }) => {
```
Comment (pongstylin) [inline src/tactics/LocalTransport.js]: Good!

File: src/tactics/LocalTransport.js [hunk 9]
```diff
@@ -148,28 +148,29 @@ export default class LocalTransport {
...
-          this._data.actions.push(...actions);
+        .on('action', ({data:actions}) => {
+          this._data.state.actions.push(...actions);
         })
-        .on('reset', ({turnId, teamId, actions}) => {
-          Object.assign(this._data, {
-            currentTurnId: turnId,
-            currentTeamId: teamId,
-            actions:       actions,
+        .on('reset', ({data}) => {
```
Comment (pongstylin) [inline src/tactics/LocalTransport.js]: @latin-programmer The 'reset' event no longer exists.  I have renamed it to the "revert" event.

File: src/tactics/core.js [hunk 10]
```diff
@@ -65,7 +65,7 @@ window.Tactics = (function () {
     },
     createLocalGame: function (stateData) {
       let transport = LocalTransport.createGame(stateData);
-      let localTeamIds = stateData.teams
+      let localTeamIds = stateData.data.teams
```
Comment (pongstylin) [inline src/tactics/core.js]: @latin-programmer This can be reverted once data is no longer nested.

**Comment by pongstylin** [comment]
@latin-programmer I finished reviewing the code changes.  So, when you're ready, please resolve the comments I had.

**Comment by latin-programmer** [comment]
@pongstylin Thank you for the review. I addressed all the issues in the code review changes. The apps are still in the same state with the new changes (so more work will need to be done), but overall I'm still good with sending what I have so far as it fixes the classic app and removes the errors I reported for the other two apps. The problems remaining are: 
 - Face off app loads but does not work (the bot may start a turn or may not, but then you cannot play after) 
 - Chaos app is missing the egg and you cannot play after.
